test/functional/p2p_invalid_messages.py

```diff
@@ -248,6 +250,19 @@ def test_oversized_headers_msg(self):
         size = MAX_HEADERS_RESULTS + 1
         self.test_oversized_msg(msg_headers([CBlockHeader()] * size), size)
 
+    def test_invalid_pow_headers_msg(self):
+        self.log.info("Test headers message with invalid proof-of-work is logged as misbehaving and disconnects peer")
+        blockheader_tip = self.nodes[0].getblockheader(self.nodes[0].getbestblockhash(), False)
```

Comment:
The node doesn't have any blocks, so you are getting the genesis block, which can't be send over p2p anyway, as the previous block is unknown.

```
received header ......: missing prev block 0000000000000000000000000000000000000000000000000000000000000000

Reply:
Yeah, in fact, since the PoW check is the first one that actually inspects `headers` data, the content of the block header doesn't matter at all for this test-case; could even be completely random bytes:
```diff
diff --git a/test/functional/p2p_invalid_messages.py b/test/functional/p2p_invalid_messages.py
index f5fe590c62..e85dc00082 100755
--- a/test/functional/p2p_invalid_messages.py
+++ b/test/functional/p2p_invalid_messages.py
@@ -29,6 +29,7 @@ from test_framework.p2p import (
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import (
     assert_equal,
+    random_bytes,
 )

 VALID_DATA_LIMIT = MAX_PROTOCOL_MESSAGE_LENGTH - 5  # Account for the 5-byte length prefix
@@ -252,8 +253,7 @@ class InvalidMessagesTest(BitcoinTestFramework):

     def test_invalid_pow_headers_msg(self):
         self.log.info("Test headers message with invalid proof-of-work is logged as misbehaving and disconnects peer")
-        blockheader_tip = self.nodes[0].getblockheader(self.nodes[0].getbestblockhash(), False)
-        blockheader_invalid_pow = from_hex(CBlockHeader(), blockheader_tip)
+        blockheader_invalid_pow = from_hex(CBlockHeader(), random_bytes(80).hex())
         blockheader_invalid_pow.rehash()
         while not blockheader_invalid_pow.hash.startswith('f'):
             blockheader_invalid_pow.nNonce += 1

```
But I agree that the test-case in its current form suggest that header would be valid _with_ correct PoW. Will take a deeper look in a bit.